pkg/skaffold/watch/triggers.go

```diff
@@ -180,3 +185,40 @@ func (t *fsNotifyTrigger) Start(ctx context.Context) (<-chan bool, error) {
 
 	return trigger, nil
 }
+
+type apiTrigger struct {
+	Trigger  chan bool
+	Interval time.Duration
```

Comment:
I don't understand why we need an interval at all for the API trigger. I'd suggest removing the logic around it.